### PR TITLE
Allow using from_iter instead of collect

### PR DESCRIPTION
**Pull request by Kerollmops**
Clippy started warning that we use from_iter instead of collect, but why not just letting us do that.

### CODE REVIEW

**Comment by Kerollmops** [comment]
bors try

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -42,4 +42,4 @@ jobs:
       - uses: actions-rs/cargo@v1
         with:
           command: clippy
-          args: --all-targets -- -D warnings
+          args: --all-targets
```
Comment (domodwyer) [inline .github/workflows/test.yml:45]: Do you still need to remove this if we have `#![allow(clippy::from_iter_instead_of_collect)]` in the appropriate places? It's probably best to deny warnings - it prevents things like unused imports, etc.

If adding it back in still causes a failure, maybe try:

```suggestion
          args: --all-targets -- -D warnings -A clippy::from_iter_instead_of_collect
```

**Comment by domodwyer** [comment]
The lint description makes sense: https://rust-lang.github.io/rust-clippy/master/#from_iter_instead_of_collect

I haven't looked but seems like we can probably fix this without a breaking change - I can take a look this too and PR to re-enable the lint over the weekend if it's still a problem 👍

**Comment by Kerollmops** [comment]
@domodwyer If you want to take some time to fix every place where we use the from_iter method where we should use collect, I would be very thankful :)

File: .github/workflows/test.yml (hunk 1 above)
Comment (Kerollmops) [reply .github/workflows/test.yml:45]: The `from_iter_instead_of_collect` lint is an unstable flag and only available on nightly, the problem is that it is complex/impossible to only allow it on nightly.

**Comment by domodwyer** [comment]
Easy! https://github.com/RoaringBitmap/roaring-rs/pull/78

I will rebase #76 after the PR is merged 👍

**Comment by Kerollmops** [comment]
Thank you for #78 but you don't need to rebase, bors will do that for us automatically :)

**Pull request closed without merging**